app/models/service_provider.rb

```diff
@@ -108,6 +108,12 @@ def logo_is_email_compatible?
     logo_url.end_with?('.png')
   end
 
+  def needs_to_reproof?(initiating_service_provider)
+    # TODO Check verification date against blackout period end
+    issuer == IdentityConfig.store.reproof_forcing_provider &&
+      initiating_service_provider&.issuer != IdentityConfig.store.reproof_forcing_provider
```

Comment:
I'm not super close to this, so apologies if this question is bad, but are there other conditions? e.g. profile level?

Reply:
I considered it, but we are doing this only for IRS and they are IAL2 with facial match, right? As I see it, once the PII enclave is added we would add the following conditions:
1. initiating_service_provider&.issuer != IdentityConfig.store.reproof_forcing_provider && profile.verified_at < PII_ENCLAVE_START_DATE
2. profile.idv_level != IPP level